gui/hud.gui

```diff
@@ -2991,6 +2991,743 @@ widget = {
+								{
+									texture = "gfx/interface/colors/gold.dds"
+								}
+
+								blockoverride "bar_color_flow"
+								{
+									texture = "gfx/interface/colors/white.dds"
+									alpha = 0.3
+								}
+							}
+						}
+
+						widget = {
+							layoutpolicy_horizontal = expanding
+							layoutpolicy_vertical = expanding
+							layoutstretchfactor_horizontal = "[GetProgressBarValueMaxOtherScaled( Add_float(FixedPointToFloat(GetPlayer.MakeScope.Var('light_void_infusion').GetValue), '(float)100'), '(float)200', '(int32)80' )]"
+						}
+					}
+
+					hbox = {
+						visible = "[LessThanOrEqualTo_int32(FixedPointToInt(GetPlayer.MakeScope.Var('light_void_infusion_level').GetValue), '(int32)-1')]"
+						widget = {
+							layoutpolicy_horizontal = expanding
+							layoutpolicy_vertical = expanding
+							layoutstretchfactor_horizontal = "[GetProgressBarValueMaxScaled( Subtract_float('(float)200', Add_float(FixedPointToFloat(GetPlayer.MakeScope.Var('light_void_infusion').GetValue), '(float)100')), '(float)200', '(int32)80' )]"
+
+							progressbar_hud_stress = {
+								visible = "[GreaterThanOrEqualTo_int32(FixedPointToInt(GetPlayer.MakeScope.Var('light_void_infusion_level').GetValue), '(int32)-3')]"
+								size = { 100% 100% }
+
+								using = Animation_ShowHide_Standard
+
+								blockoverride "bar_color"
+								{
+									texture = "gfx/interface/colors/black_purple.dds"
+								}
+
+								blockoverride "bar_color_flow"
+								{
+									texture = "gfx/interface/colors/gold.dds"
+									alpha = 0.7
+								}
+							}
+
+							progressbar_hud_stress = {
+								visible = "[EqualTo_int32(FixedPointToInt(GetPlayer.MakeScope.Var('light_void_infusion_level').GetValue), '(int32)-2')]"
+								size = { 100% 100% }
+
+								using = Animation_ShowHide_Standard
+
+								blockoverride "bar_color"
+								{
+									texture = "gfx/interface/colors/dark_purple.dds"
+								}
+
+								blockoverride "bar_color_flow"
+								{
+									texture = "gfx/interface/colors/white.dds"
+									alpha = 0.3
+								}
+							}
+
+							progressbar_hud_stress = {
+								visible = "[EqualTo_int32(FixedPointToInt(GetPlayer.MakeScope.Var('light_void_infusion_level').GetValue), '(int32)-1')]"
+								size = { 100% 100% }
+
+								using = Animation_ShowHide_Standard
+
+								blockoverride "bar_color"
+								{
+									texture = "gfx/interface/colors/purple.dds"
+								}
+
+								blockoverride "bar_color_flow"
+								{
+									texture = "gfx/interface/colors/white.dds"
+									alpha = 0.3
+								}
+							}
+						}
+
+						widget = {
+							layoutpolicy_horizontal = expanding
+							layoutpolicy_vertical = expanding
+							layoutstretchfactor_horizontal = "[GetProgressBarValueMaxOtherScaled( Subtract_float('(float)200', Add_float(FixedPointToFloat(GetPlayer.MakeScope.Var('light_void_infusion').GetValue), '(float)100')), '(float)200', '(int32)80' )]"
+						}
+					}
```

Comment:
You should use `type` and `template` to make the code readable. This amount of lines isn't normal and I see it's repetative.
And it will be super-duper if you place them in `gui\shared`.

Reply:
That part is copy pasted from the vanilla stress bar, I think I can slim some of it down, but most lines change every time they're used.